src/Diagnostics.ModelsAndUtils/Models/GeoMaster/NetworkResourceProviderEntities.cs

```diff
@@ -59,64 +127,123 @@ public int CompareTo(NetworkResourceProviderNsgRule other)
         }
     }
 
-    public class NetworkResourceProviderNsg
+    public class EffectiveRoute
     {
-        public string ResourceGuid { get; set; }
-        public IEnumerable<NetworkResourceProviderNsgRule> SecurityRules { get; set; }
-        public IEnumerable<NetworkResourceProviderNsgRule> DefaultSecurityRules { get; set; }
+        public string Name { get; set; }
+
+        public RouteSource Source { get; set; }
+
+        public RouteStatus Status { get; set; }
+
+        public string[] AddressPrefixes { get; set; }
+
+        public EffectiveNextHop EffectiveNextHop { get; set; }
     }
 
-    public class NetworkResourceProviderSubnetNsgRef
+    [JsonConverter(typeof(StringEnumConverter))]
+    public enum RouteSource
     {
-        public string Id { get; set; }
-    }
+        User,
+        VPNGateway,
+        Default
+    };
 
-    public class NetworkResourceProviderIpConfigurationRef
+    [JsonConverter(typeof(StringEnumConverter))]
+    public enum RouteStatus
     {
-        public string Id { get; set; }
-    }
+        Active, Invalid
+    };
 
-    public class NetworkResourceProviderServiceEndpointRef
+    [JsonConverter(typeof(StringEnumConverter))]
+    public enum RouteType
     {
-        public string ProvisioningState { get; set; }
-        public string Service { get; set; }
-        public string[] Locations { get; set; }
+        VPNGateway,
+        VNETLocal,
+        Internet,
+        VirtualAppliance,
+        Null
+    };
+
+    public class EffectiveNextHop
+    {
+        public RouteType Type { get; set; }
+
+        public string[] IpAddresses { get; set; }
     }
+    
 
-    public class NetworkResourceProviderSubnetDelegation
+    public class EffectiveRouteTable
     {
-        public string ProvisioningState { get; set; }
-        public string ServiceName { get; set; }
-        public IEnumerable<string> Actions { get; set; }
+        public EffectiveRoute[] EffectiveRoutes { get; set; }
     }
 
-    public class NetworkResourceProviderSubnet
+    /// <summary>
+    /// Message envelope that contains the common Azure resource manager properties and the resource provider specific content.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class ResponseMessageEnvelope<T>
```

Comment:
Instead of using this class you can use [GeomasterResponse.cs](https://github.com/Azure/Azure-AppServices-Diagnostics/blob/master2/src/Diagnostics.ModelsAndUtils/Models/GeoMaster/GeoMasterResponse.cs) class